tests/error/nested_errors/reassign_capture_3.err

```diff
@@ -1,7 +1,12 @@
-Guppy compilation failed. Error in file $FILE:11
+Error: Illegal assignment (at $FILE:11:12)
+   | 
+ 9 | 
+10 |         def baz() -> None:
+11 |             y += 2
+   |             ^^^^^^ Variable `y` may not be assigned to since `y` is captured
+   |                    from an outer scope
+   | 
+ 6 |     y = x + 1
+   |     - `y` defined here
```

Comment:
Yes, that's a good idea. However, we'll need to be careful because the grammar might break in some cases when reordering:

```
Error: Linearity violation (at $FILE:21:8)
   | 
19 | def test(q: qubit @owned) -> None:
20 |     use(q)
21 |     foo(q)
   |         ^ Variable `q` with linear type `qubit` cannot be borrowed
   |           ...
   | 
20 |     use(q)
   |         - since it was already consumed here
```

We'd need two version of the span labels depending on which order they are outputted. I created #608 to collect these improvement ideas